python/lsst/daf/butler/core/webdavutils.py

```diff
@@ -0,0 +1,167 @@
+# This file is part of daf_butler.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+__all__ = ("getHttpSession", "isWebdavEndpoint", "webdavCheckFileExists",
+           "folderExists", "webdavDeleteFile", "getFileURL")
+
+import os
+import requests
+import logging
+from requests.structures import CaseInsensitiveDict
+
+from typing import (
+    Optional,
+    Tuple,
+    Union,
+)
+
+from .location import ButlerURI, Location
+
+log = logging.getLogger(__name__)
+
+
+def getHttpSession() -> requests.Session:
+    """Create a requests.Session pre-configured with environment variable data
+
+    Returns
+    -------
+    s : `requests.Session`
+        An http session used to execute requests.
+
+    Notes
+    -----
+    The WEBDAV_AUTH_METHOD must be set to obtain a session.
+    Depending on the chosen method, additional
+    environment variables are required:
+
+    X509: must set WEBDAV_PROXY_CERT
+    (path to proxy certificate used to authenticate requests)
+
+    TOKEN: must set WEBDAV_BEARER_TOKEN
+    (bearer token used to authenticate requests, as a single string)
+
+    NB: requests will read CA certificates in REQUESTS_CA_BUNDLE
+    It must be manually exported according to the system CA directory.
+    """
+    s = requests.Session()
```

Comment:
Can you use `session` as the variable name to be more in line with other places here.